octue/resources/child.py

```diff
@@ -0,0 +1,18 @@
+from octue.resources.service import Service
+
+
+class Child:
```

Comment:
This seems like an unnecessary level of abstraction. Why not just have a method defined on the `Service` class?
```
def ask_and_wait(self):
        """ Ask this child service a question (i.e. send it some input value and/or a manifest and wait for it to run an analysis
        on them and return the output values). The input values given must adhere to the Twine file of the child.
        """

        subscription = self.ask(self.id, input_values, input_manifest)
        return self.wait_for_answer(subscription, timeout)

Reply:
The point of this class is to modify the semantics of asking a question. Rather than a service asking another service a question e.g.
```
my_service.ask(another_service, question)
```

the `Child` instance *representing* a service allows you to ask it a question e.g.
```
child.ask(question)
```

I added this as we thought a Scientist wouldn't by default think of themselves as having a service to ask questions with, but would instead think of just asking the children they have questions.

Reply:
Ah, I get it! yes, ok gotcha.